frame/staking-rewards/src/lib.rs

```diff
@@ -599,8 +626,118 @@ pub mod pallet {
 					.try_insert(*asset_id, inflation)
 					.map_err(|_| Error::<T>::ReductionConfigProblem)?;
 			}
+
 			Ok((rewards_btree_map, reductions))
 		}
+
+		pub(crate) fn reward_acumulation_hook_reward_update_calculation(
```

Comment:
I prefixed the name with `reward_acumulation_hook` since the function should only be used within that hook (I had to separate it out for benchmarking) but I'm not partial to it